plugins/woocommerce/src/Admin/PluginsHelper.php

```diff
@@ -618,4 +644,258 @@ public static function maybe_enqueue_scripts_for_notices_in_plugins() {
+			return sprintf(
+				$messages[ $message_key ],
+				esc_attr( $subscription['product_name'] ),
+				esc_attr( $expiry_date ),
+				esc_url( self::WOO_SUBSCRIPTION_PAGE_URL ),
+				esc_attr( $price ),
+			);
+		}
+
+		return '';
+	}
+
+	/**
+	 * Get formatted notice information for expiring subscription.
+	 *
+	 * @param boolean $allowed_link whether the notice description should include a link.
+	 * @return array notice information.
+	 */
+	public static function get_expiring_subscription_notice( $allowed_link = true ) {
+		if ( ! WC_Helper::is_site_connected() ) {
+			return array();
+		}
+
+		if ( ! self::$can_show_expiring_subs_notice ) {
+			return array();
+		}
+
+		if ( ! self::should_show_notice( self::DISMISS_EXPIRING_SUBS_NOTICE ) ) {
+			return array();
+		}
+
+		$subscriptions          = WC_Helper::get_subscriptions();
+		$expiring_subscriptions = array_filter(
+			$subscriptions,
+			function ( $sub ) {
+				return $sub['expiring'] && ! $sub['autorenew'];
+			},
+		);
+
+		if ( ! $expiring_subscriptions ) {
+			return array();
+		}
+
+		$total_expiring_subscriptions = count( $expiring_subscriptions );
+
+		$message = self::get_subscriptions_message(
+			$subscriptions,
+			$expiring_subscriptions,
+			$total_expiring_subscriptions,
+			array(
+				/* translators: 1) product name 2) expiry date 3) URL to My Subscriptions page */
+				'single_manage'           => __( 'Your subscription for <strong>%1$s</strong> expires on %2$s. <a href="%3$s">Enable auto-renewal</a> to continue receiving updates and streamlined support.', 'woocommerce' ),
+				/* translators: 1) product name 2) expiry date 3) URL to My Subscriptions page */
+				'multiple_manage'         => __( 'One of your subscriptions for <strong>%1$s</strong> expires on %2$s. <a href="%3$s">Enable auto-renewal</a> to continue receiving updates and streamlined support.', 'woocommerce' ),
+				/* translators: 1) total expiring subscriptions 2) URL to My Subscriptions page */
+				'different_subscriptions' => __( 'You have <strong>%1$s Woo extension subscriptions</strong> expiring soon. <a href="%2$s">Enable auto-renewal</a> to continue receiving updates and streamlined support.', 'woocommerce' ),
+			)
+		);
+
+		return array(
+			'description' => $allowed_link ? $message : preg_replace( '#<a.*?>(.*?)</a>#i', '\1', $message ),
+			'button_text' => __( 'Enable auto-renewal', 'woocommerce' ),
+			'button_link' => self::WOO_SUBSCRIPTION_PAGE_URL,
+		);
+	}
+
+	/**
+	 * Get formatted notice information for expired subscription.
+	 *
+	 * @param boolean $allowed_link whether the notice description should include a link.
+	 * @return array notice information.
+	 */
+	public static function get_expired_subscription_notice( $allowed_link = true ) {
+		if ( ! WC_Helper::is_site_connected() ) {
+			return array();
+		}
+
+		if ( ! self::should_show_notice( self::DISMISS_EXPIRED_SUBS_NOTICE ) ) {
+			return array();
+		}
+
+		$subscriptions         = WC_Helper::get_subscriptions();
```

Comment:
note that this gets connected user's subscriptions (does not consider the site it's connected from)